### ISSUE TITLE
Do not parse files contained within directories marked as ignored.

### ISSUE THREAD

**Issue by matt-usurp** [hsz/idea-gitignore#125]
When opening a project where a dependency manager is available (PHP and Composer) the ignore files are indexed within the dependency directory that is marked as ignored by the projects root ignore.

## In more detail

You have a project such as Symfony 2 using PHP's community based dependency manager composer. These files are downloaded to a directory within your project which is then ignored using (git) ignore files. Using Ideas "exclude directory" feature is not viable here as the reason you have the dependencies locally is so you have code completion. Without the ignore plugin this works fine. The issue arises when the ignore plugin indexes all the ignore files in the project **including** the ones contained in directories that are marked within other ignore files.

A side note; I am using PHPStorm 8.0.3 with the latest version of the plugin available.

## Example

contents of **.gitignore**
```
vendor
composer.phar
``` 

The following messages are shown
```
Ignore rules indexing
- vendor/symfony/http-foundation/.ignore
- vendor/symfony/framework/.ignore
- vendor/symfony/blah/.ignore
- and so on
```

The expected behaviour is that the root ignore file is parsed and then the directory tree is traversed from there. In this case the ignore file would see that `vendor` is ignored and not bother indexing the 25+ other ignore files contained inside.

There is a similar ticket #78 which I thought was going to cover this but it is marked as closed and the conversation within it was not as expected.

**Comment by hsz** [comment]
@matt-usurp Thank you for reporting.

Which version of plugin do you use ?

**Comment by matt-usurp** [comment]
The PHPStorm plugin manager says 1.1.2

**Comment by hsz** [comment]
@matt-usurp  Fix has been already released with version v1.1.3 and is available for download.

**Comment by matt-usurp** [comment]
@hsz Thanks! Project loading times have been cut to a fraction. Much appreciated!
